xbmc/addons/gui/GUIDialogAddonInfo.cpp

```diff
@@ -337,7 +337,17 @@ void CGUIDialogAddonInfo::OnInstall()
   if (!m_item->HasAddonInfo())
     return;
 
-  if (m_localAddon)
+  const auto& itemAddonInfo = m_item->GetAddonInfo();
+
+  const std::string& addonId = itemAddonInfo->ID();
+  const std::string& origin = itemAddonInfo->Origin();
+  const AddonVersion& version = itemAddonInfo->Version();
+
+  const AddonRepoUpdateMode updateMode =
+      CAddonSystemSettings::GetInstance().GetAddonRepoUpdateMode();
+
+  if (m_localAddon && (m_localAddon->Origin() != origin) &&
```

Comment:
No need to unconditionally calculate `updateMode`. It's only evaluated if `m_localAddon->Origin() != origin`.

Reply:
```c++
const auto& itemAddonInfo = m_item->GetAddonInfo();
const std::string& origin = itemAddonInfo->Origin();

if (m_localAddon && (m_localAddon->Origin() != origin))
{
  const AddonRepoUpdateMode updateMode =
      CAddonSystemSettings::GetInstance().GetAddonRepoUpdateMode();
  if (updateMode != AddonRepoUpdateMode::ANY_REPOSITORY)
  {
    const std::string& header = ...
  }
}

const std::string& addonId = itemAddonInfo->ID();
const AddonVersion& version = itemAddonInfo->Version();
```